syntax = "proto2";

package pb_test;

message Person {
  optional string name = 1;
  optional uint32 id = 2;
  optional string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    optional uint32 number = 1;
    optional PhoneType type = 2 [default = HOME];
  }
  repeated float scores = 16 [packed=true];
  repeated PhoneNumber phones = 4;  
  optional string address = 8;
  optional string nationality = 9;
  optional string bitrthday = 10;
  optional string creationDate = 11;
  optional string updateDate = 12;
  optional string selfIntroduction = 13;
}
